pkg/controller/apiextensions/composite/composed/api.go

```diff
@@ -114,10 +115,17 @@ func (cdf *APIConnectionDetailsFetcher) Fetch(ctx context.Context, cd resource.C
 	// iteration.
 	s := &corev1.Secret{}
 	nn := types.NamespacedName{Namespace: sref.Namespace, Name: sref.Name}
-	if err := cdf.client.Get(ctx, nn, s); resource.IgnoreNotFound(err) != nil {
+	err := cdf.client.Get(ctx, nn, s)
```

Comment:
Nit: This could be abbreviated:

```go
if err := ...; err != nil {
    return nil, errors.Wrap(resource.IgnoreNotFound(err), errGetSecret)
}
```